### PR TITLE
Update InvalidateCacheOutputAttribute.cs

### PR DESCRIPTION
**Pull request by oicguy**
Overriding the `OnResultExecutionAsync` method allows us to use the StatusCode set through using the ActionResult methods, falling back to the original test when unavailable.

### CODE REVIEW

File: AspNetCore.CacheOutput/InvalidateCacheOutputAttribute.cs [hunk 1]
```diff
@@ -26,17 +26,27 @@ public InvalidateCacheOutputAttribute(string methodName, Type controllerType, Ty
             this.cacheKeyGeneratorType = cacheKeyGeneratorType;
         }
 
-        public override async Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
+        private bool IsFailure(int statusCode)
```
Comment (Iamcerba) [inline AspNetCore.CacheOutput/InvalidateCacheOutputAttribute.cs:29]: 1. Reorder public-private methods, private at the bottom.
2. Could you come up with a better name? 300 range, for example, doesn't necessarily mean failure.

File: AspNetCore.CacheOutput/InvalidateCacheOutputAttribute.cs [hunk 2]
```diff
@@ -26,17 +26,27 @@ public InvalidateCacheOutputAttribute(string methodName, Type controllerType, Ty
...
-            )
+            return !(
+                    statusCode >= (int)HttpStatusCode.OK &&
+                    statusCode < (int)HttpStatusCode.Ambiguous
+                );
+        }
+
+        public override async Task OnResultExecutionAsync(ResultExecutingContext context, ResultExecutionDelegate next)
+        {
+            await base.OnResultExecutionAsync(context, next);
```
Comment (Iamcerba) [inline AspNetCore.CacheOutput/InvalidateCacheOutputAttribute.cs:39]: Could we keep old logic and just change OnActionExecutionAsync to OnResultExecutionAsync?

File: AspNetCore.CacheOutput/InvalidateCacheOutputAttribute.cs [hunk 3]
```diff
@@ -26,17 +26,27 @@ public InvalidateCacheOutputAttribute(string methodName, Type controllerType, Ty
...
+
+#if NETCOREAPP2_0 || NETCOREAPP2_1
```
Comment (Iamcerba) [inline AspNetCore.CacheOutput/InvalidateCacheOutputAttribute.cs:41]: .NET Core 2.2 also should be handled.

File: AspNetCore.CacheOutput/InvalidateCacheOutputAttribute.cs [hunk 4]
```diff
@@ -26,17 +26,27 @@ public InvalidateCacheOutputAttribute(string methodName, Type controllerType, Ty
...
+            var isFailure = IsFailure(context.HttpContext.Response.StatusCode);
+#else
+            var result = context.Result as IStatusCodeActionResult;
```
Comment (Iamcerba) [inline AspNetCore.CacheOutput/InvalidateCacheOutputAttribute.cs:44]: The reference couldn't be resolved.

File: AspNetCore.CacheOutput/InvalidateCacheOutputAttribute.cs (hunk 3 above)
Comment (oicguy) [reply AspNetCore.CacheOutput/InvalidateCacheOutputAttribute.cs:41]: .Net Core 2.2+ is handled by the else condition. 2.0 and 2.1 are the only ones that need to be specially cared for

File: AspNetCore.CacheOutput/InvalidateCacheOutputAttribute.cs (hunk 1 above)
Comment (oicguy) [reply AspNetCore.CacheOutput/InvalidateCacheOutputAttribute.cs]: Done

File: AspNetCore.CacheOutput/InvalidateCacheOutputAttribute.cs (hunk 4 above)
Comment (oicguy) [reply AspNetCore.CacheOutput/InvalidateCacheOutputAttribute.cs:37]: Sorry about that. I've fixed it.

File: AspNetCore.CacheOutput/InvalidateCacheOutputAttribute.cs (hunk 3 above)
Comment (oicguy) [reply AspNetCore.CacheOutput/InvalidateCacheOutputAttribute.cs:34]: So the IStatusCodeActionResult interface is only available for .Net Core 2.2+.  As a result, I only need to handle the the earlier ones. Going forward, the new code handles the rest.

File: AspNetCore.CacheOutput/InvalidateCacheOutputAttribute.cs (hunk 2 above)
Comment (oicguy) [reply AspNetCore.CacheOutput/InvalidateCacheOutputAttribute.cs:32]: By 'old logic' I can only assume you mean the test for Response not being null. I can adjust the code to get the equivelent of that but using the exact same code would be problematic.

**Comment by Iamcerba** [comment]
@oicguy appreciate for contribution :)

**Comment by oicguy** [comment]
No problem!

**Pull request merged**